internal/controller/pie/pieprobe_controller_test.go

```diff
@@ -148,6 +161,108 @@ var _ = Describe("PieProbe controller", func() {
 		time.Sleep(100 * time.Millisecond)
 	})
 
+	It("should create PVCs with the capacity specified in the PieProbe resource", func() {
+		By("checking the default PVC's capacity is 100Mi")
+		Eventually(func(g Gomega) {
+			var pvcList corev1.PersistentVolumeClaimList
+			err := k8sClient.List(ctx, &pvcList, client.MatchingLabels(map[string]string{
+				"storage-class": "sc",
+				"node":          "192.168.0.1",
+			}))
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(len(pvcList.Items)).Should(Equal(1))
+
+			capacity, ok := pvcList.Items[0].Spec.Resources.Requests.Storage().AsInt64()
+			g.Expect(ok).To(BeTrue())
+			g.Expect(capacity).Should(Equal(int64(100 * 1024 * 1024)))
+
+			var cronJobList batchv1.CronJobList
+			err = k8sClient.List(ctx, &cronJobList)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(len(cronJobList.Items)).Should(Equal(3))
+			for _, cronJob := range cronJobList.Items {
+				if !strings.HasPrefix(cronJob.GetName(), "provision-") {
+					continue
+				}
+				g.Expect(
+					cronJob.Spec.JobTemplate.Spec.Template.Spec.Volumes[0].VolumeSource.Ephemeral.VolumeClaimTemplate.
+						Spec.Resources.Requests[corev1.ResourceStorage].Equal(*resource.NewQuantity(100*1024*1024, resource.BinarySI)),
+				).To(BeTrue())
```

Comment:
I believe you can write it more simply in the form of `Expect(ACTUAL).To(Equal(EXPECTED))`.


cf. https://[onsi.github.io/gomega/#making-assertions](https://onsi.github.io/gomega/#making-assertions)

Reply:
Actually, rewriting this code as follows doesn't work:

```go
g.Expect(
	cronJob.Spec.JobTemplate.Spec.Template.Spec.Volumes[0].VolumeSource.Ephemeral.VolumeClaimTemplate.
		Spec.Resources.Requests[corev1.ResourceStorage],
).To(Equal(*resource.NewQuantity(100*1024*1024, resource.BinarySI)))
```

```
  [FAILED] Timed out after 10.001s.
  The function passed to Eventually failed at /home/rbanno/pie/internal/controller/pie/pieprobe_controller_test.go:190 with:
  Expected
      <resource.Quantity>: {
          i: {value: 104857600, scale: 0},
          d: {Dec: nil},
          s: "100Mi",
          Format: "BinarySI",
      }
  to equal
      <resource.Quantity>: {
          i: {value: 104857600, scale: 0},
          d: {Dec: nil},
          s: "",
          Format: "BinarySI",
      }
  In [It] at: /home/rbanno/pie/internal/controller/pie/pieprobe_controller_test.go:192 @ 08/07/24 04:37:28.749
```

This is because `resource.Quantity`'s `Equal` isn't a simple `reflect.DeepEqual`; it also considers the semantics. There can be multiple inner representations of `resource.Quantity` for one same quantity, and `resource.Quantity`'s `Equal` needs to return `true` for all of them. However, Gomega's `Equal` doesn't call `resource.Quantity`'s `Equal` but calls `reflect.DeepEqual`, so it doesn't work properly here.